src/Package.php

```diff
@@ -152,6 +170,7 @@ private function deduceTrackingCode(): Package
         }
 
         if (!empty($carrier_code)) {
+            $this->tracking_url = Carrier::buildTrackingUrl($carrier_code, $matches[0]);;
```

Comment:
hey @bell2041, apologies for the late reply - thank you for the updates. Can you remove the extra `;` at the end of this line?

Also, I'm wondering if we should remove this line of code (and the `tracking_url` property in this Package class) and instead update your getTrackingUrl() method in the Package class to grab the url from the Carrier object since it has all the info it needs. Something like this:

```
    public function getTrackingUrl(): string
    {
        return $this->carrier->tracking_url;
    }
```

To do this, I think you'd need to pass in the $tracking_code as a second param to the Carrier class and auto call buildTrackingUrl() within the constructor. Something like this for the code below:

```
$this->carrier = new Carrier($carrier_code, $tracking_code);
```

I think that'd clean things up a bit more - what do you think?